src/lib/mongoBackend/mongoUnsubscribeContext.cpp

```diff
@@ -127,20 +127,22 @@ HttpStatusCode mongoUnsubscribeContext
   //
   // Removing subscription from mongo subscription cache
   //
-  LM_T(LmtSubCache, ("removing subscription '%s' (tenant '%s') from mongo subscription cache",
-                     requestP->subscriptionId.get().c_str(),
-                     tenant.c_str()));
+  if (!noCache)
+  {
+    LM_T(LmtSubCache, ("removing subscription '%s' (tenant '%s') from mongo subscription cache",
+                       requestP->subscriptionId.get().c_str(),
+                       tenant.c_str()));
 
-  cacheSemTake(__FUNCTION__, "Removing subscription from cache");
+    cacheSemTake(__FUNCTION__, "Removing subscription from cache");
 
-  CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
-
-  if (cSubP != NULL)
-  {
-    subCacheItemRemove(cSubP);
-  }
+    CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
 
+    if (cSubP != NULL)
+    {
+      subCacheItemRemove(cSubP);
+    }
   cacheSemGive(__FUNCTION__, "Removing subscription from cache");
+  }
   reqSemGive(__FUNCTION__, "ngsi10 unsubscribe request", reqSemTaken);
```

Comment:
Code style:

```suggestion
    cacheSemGive(__FUNCTION__, "Removing subscription from cache");
  }
  
  reqSemGive(__FUNCTION__, "ngsi10 unsubscribe request", reqSemTaken);
```

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/58469edb28b166b8fbf80c01f7e91f7f0fd553b6